cpu/stm32_common/periph/flashpage.c

```diff
@@ -201,3 +206,24 @@ void flashpage_write(int page, const void *data)
         flashpage_write_raw(page_addr, data, FLASHPAGE_SIZE);
     }
 }
+
+#ifdef MODULE_PERIPH_EEPROM
```

Comment:
I think this files is missing the `#include "periph/eeprom.h"`? But would also suggest to not add this to flashpage driver but rather have a separate file.

Reply:
> I think this files is missing the #include "periph/eeprom.h"?

Agreed

> But would also suggest to not add this to flashpage driver but rather have a separate file.

How do you share code between different periph ? Here I'm reusing the lock/unlock functions from flashpage periph, since they are the same.

Reply:
@smlng, I moved the _lock/_unlock into a shared file (between flashpage and eeprom periph drivers) and provided the eeprom peripheral driver into a separate file: now there's no dependency between flashpage and eeprom (for STMs).